src/facet.js

```diff
@@ -38,9 +38,10 @@ class Facet extends Mark {
     }
     for (let i = 0; i < this.marks.length; ++i) {
       const mark = this.marks[i];
-      const markFacets = mark.facet == null ? mark.data === this.data ? facetsIndex : undefined
-        : mark.facet === "exclude" ? facetsIndex.map(facet => Uint32Array.from(difference(index, facet)))
-        : mark.facet === true ? facetsIndex
+      const facet = mark.facet && keyword(mark.facet, "facet", ["include", "exclude"]);
+      const markFacets = facet === undefined ? mark.data === this.data ? facetsIndex : undefined
+        : facet === "include" ? facetsIndex
+        : facet === "exclude" ? facetsIndex.map(f => Uint32Array.from(difference(index, f)))
```

Comment:
I think we probably shouldn’t bother with this optimization, but: this could be extracted out of the for loop using a lazily-constructed variable so that it can be shared across multiple excluded marks.

Reply:
it's cheap so I added it too